reactor-core/src/main/java/reactor/core/publisher/MonoProcessor.java

```diff
@@ -161,22 +203,21 @@ O block(Runnable spinObserver) {
 			if (!isPending()) {
 				return peek();
 			}
-			else if(subscription == null) {
-				getOrStart();
-			}
+
+			connect();
 
 			try {
-				long endState = waitStrategy.waitFor(STATE_SUCCESS_VALUE, this, spinObserver);
+				long endState = waitStrategy.waitFor(3L, this, spinObserver);
```

Comment:
just put a comment along each int magic number (here and in the `switch`) to explicitly tell what state it correspond to, so that future us don't get crazy reading that code 😛